### DISCUSSION TITLE
Macro to act on selected text?

### DISCUSSION THREAD

**Discussion by bernatagulloesbrina** [TabularEditor/TabularEditor3#1073] [Ideas]
Would it be possible to make a macro to act on selected text? 
my use case would be wrapping things inside a calculate, or calculate table function 
you select the expression, click on the macro and magically the text is now wrapped around with the calculate/calculate table funciton, and the cursor is ready to add some filters! 
You should be able to use them while editinc measures, but also writting DAX queries and maybe even c# scripts? That would make the creation of dax scripts faster! 

PhraseExpander does something like that

regards!

**Comment by otykier** [comment]
We do have something like this already on the backlog. Imagine some kind of configurable DAX code snippets that you can assign shortcuts to. Specifically for wrapping an expression in CALCULATE/CALCULATETABLE, this is actually something that most users would probably find useful, so this should be a native feature in my opinion.

Macros are not designed for this, however, as they are supposed to interact with the TOM, not the DAX editor.

**Comment by bernatagulloesbrina** [comment]
yes, is a different kind of programming -- whatever suits you best
I'm thinking on trying it up in phraseExpander, although it does not work
well in programs with intellisense

I wonder if there's any opensource project that (with some extra work)
could be installed as Tabular Editor 3 plugin




El mié, 3 nov 2021 a las 15:29, Daniel Otykier ***@***.***>)
escribió:

> We do have something like this already on the backlog. Imagine some kind
> of configurable DAX code snippets that you can assign shortcuts to.
> Specifically for wrapping an expression in CALCULATE/CALCULATETABLE, this
> is actually something that most users would probably find useful, so this
> should be a native feature in my opinion.
>
> Macros are not designed for this, however, as they are supposed to
> interact with the TOM, not the DAX editor.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/TabularEditor/TabularEditor3/issues/283#issuecomment-959249102>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJUUXHK7HVNMI22ZS7ALPTDUKFBNFANCNFSM5HJBZRGQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 

Bernat Agulló

Partner & Data Enthusiast

***@***.***

+34 637 93 42 76

Balmes, 177

08006 Barcelona
